app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -695,5 +695,33 @@ def set_vm_interfaces_attributes(vm, vm_attrs)
       end
       vm_attrs
     end
+
+    def get_template_volumes_by_name(template_name)
+      template = template(template_name)
+      return {} unless template
+      return {} if template.volumes.nil?
+      template.volumes.index_by(&:name)
+    end
+
+    def volume_to_attributes(volume, template_volumes)
+      {
+        size_gb: (volume.size.to_i / 1.gigabyte),
+        storage_domain: volume.storage_domain,
+        preallocate: (volume.sparse == 'true') ? '0' : '1',
+        wipe_after_delete: volume.wipe_after_delete,
+        interface: volume.interface,
+        bootable: volume.bootable,
+        id: template_volumes.fetch(volume.name, nil)&.id,
+      }
+    end
+
+    def set_vm_volumes_attributes(vm, vm_attrs)
+      template_volumes = vm_attrs[:template] ? get_template_volumes_by_name(vm_attrs[:template]) : {}
```

Comment:
you can add 
```ruby
return {} unless template_name
```
as first step inside `get_template_volumes_by_name` instead of this ternary condition